src/com/yetanalytics/lrs_admin_ui/handlers.cljs

```diff
@@ -509,6 +511,67 @@
  (fn [_ _]
    {:dispatch [:new-account/set-password (pass/pass-gen 12)]}))
 
+(re-frame/reg-event-db
+ :update-password/set-old-password
+ global-interceptors
+ (fn [db [_ password]]
+   (assoc-in db [::db/update-password :old-password] password)))
+
+(re-frame/reg-event-db
+ :update-password/set-new-password
+ global-interceptors
+ (fn [db [_ password]]
+   (assoc-in db [::db/update-password :new-password] password)))
+
+(re-frame/reg-event-db
+ :update-password/clear
+ global-interceptors
+ (fn [db _]
+   (assoc db ::db/update-password {:old-password nil
+                                   :new-password nil})))
+
+(re-frame/reg-event-fx
+ :update-password/generate-password
+ global-interceptors
+ (fn [_ _]
+   {:dispatch [:update-password/set-new-password (pass/pass-gen 12)]}))
+
+(re-frame/reg-event-fx
+ :update-password/update-password!
+ global-interceptors
+ (fn [{{server-host ::db/server-host
+        :as db} :db} _]
+   (let [update-password (::db/update-password db)]
+     (if (valid? ::input/valid-update-password update-password)
+       {:http-xhrio {:method          :put
+                     :uri             (httpfn/serv-uri
+                                       server-host
+                                       "/admin/account/password")
+                     :response-format (ajax/json-response-format {:keywords? true})
+                     :params          update-password
+                     :format          (ajax/json-request-format)
+                     :on-success      [:update-password/update-success]
+                     :on-failure      [:update-password/update-error]
+                     :interceptors    [httpfn/add-jwt-interceptor]}}
+       {:fx         [[:dispatch [:notification/notify true
+                                 "Old or new password did not meet requirements."]]]}))))
```

Comment:
If you're going to let any string go through as the old password, then this should just say "New password did not meet requirements."

Reply:
https://github.com/yetanalytics/lrs-admin-ui/pull/43/commits/19d7697fb84f732a683acea4f6a6b82a071cd996

Reply:
Addressed in https://github.com/yetanalytics/lrs-admin-ui/pull/43/commits/19d7697fb84f732a683acea4f6a6b82a071cd996